source/xcb.c

```diff
@@ -528,10 +528,25 @@ static int monitor_active_from_id ( int mon_id, workarea *mon )
         xcb_get_property_cookie_t awc;
         awc = xcb_ewmh_get_active_window ( &xcb->ewmh, xcb->screen_nbr );
         if ( xcb_ewmh_get_active_window_reply ( &xcb->ewmh, awc, &active_window, NULL ) ) {
+            // Failure case, what if there's no active window?
+            if (active_window == 0) {
+                if ( pointer_get ( root, &x, &y ) ) {
+                    monitor_dimensions ( x, y, mon );
+                    return TRUE;
+                }
+                // This is our give up point.
+                return FALSE;
+            }
             // get geometry.
             xcb_get_geometry_cookie_t c  = xcb_get_geometry ( xcb->connection, active_window );
             xcb_get_geometry_reply_t  *r = xcb_get_geometry_reply ( xcb->connection, c, NULL );
             if ( r ) {
+                // We don't need to translate.
```

Comment:
We need translation, in I3 parent window is used to draw border, r->x, and r->y are relative to this window, not the root window.
Under my I3 this is equal to the border width.

Reply:
I added some print statements, https://github.com/ironyman/rofi/compare/next...ironyman:debugstuff

```
monitor_active_from_id -4
active window 14680074
active window geometry r 2565 37
translated geometry t 5131 75
mon 0 0 5120 1440
monitor_active_from_id -4
active window 14680074
active window geometry r 2565 37
translated geometry t 5131 75
mon 0 0 5120 1440
```

Note the `translated geometry t 5131 75`. 

I'm not sure what's supposed to happen in this case. What do you think?

Reply:
To give more context, my setup is two monitors side by side, each has width 2560. The active window is in the right one, hence the `2565 37`.

Reply:
This is odd. I have almost similar setup but 1 full hd other 1600x1200.

For me it returns 4x4 for window geometry (the i3 border is 4 pixels). Translated is then right.
I should try to setup xmonad.

Reply:
```
monitor_active_from_id -4
active window 67108865
active window geometry r 2 2
translated geometry t 1939 43
mon 1920 0 1600 1200
monitor_active_from_id -4
active window 67108865
active window geometry r 2 2
translated geometry t 1939 43
mon 1920 0 1600 1200
```